internal/js_parser/js_parser.go

```diff
@@ -13812,39 +13813,19 @@ func Parse(log logger.Log, source logger.Source, options Options) (result js_ast
 		}
 	}
 
-	if options.spinxHotReact && options.jsx.Parse {
+	if options.spinxHotReact && options.jsx.Parse && strings.HasSuffix(source.PrettyPath, ".tsx") {
```

Comment:
Checking the file extension in every loop iteration was bone-headed. There are better way to do this than a string comparison, but it's better than it was.